File: backend/app/validators/file_validator.py
async def validate_image(file):
    if file.content_type not in ["image/jpeg", "image/png"]:
        raise ValueError("Only JPG and PNG images are allowed")

    file_bytes = await file.read()
    await file.seek(0)

    if len(file_bytes) > 2 * 1024 * 1024:
        raise ValueError("File size must be less than 2MB")

    return file_bytes

File: frontend/src/validators/fileValidator.ts
export const validateImage = (file: File) => {
    const allowedTypes = ["image/jpeg", "image/png"];    // to manage supabase storage efficiently

    if (!allowedTypes.includes(file.type)) {
        throw new Error("Only JPG and PNG images are allowed");
    }

    if (file.size > 2 * 1024 * 1024) {
        throw new Error("Image must be less than 2MB");
    }
};

export const MAX_RULES_FILE_SIZE_BYTES = 10 * 1024 * 1024; // 10MB

export const validateRulesFile = (file: File) => {
    if (file.size > MAX_RULES_FILE_SIZE_BYTES) {
        throw new Error("File size must be 10MB or less");
    }
};